src/Storages/StorageDistributed.cpp

```diff
@@ -1875,4 +1875,19 @@ void registerStorageDistributed(StorageFactory & factory)
     });
 }
 
+bool StorageDistributed::initializeDiskOnConfigChange(const std::set<String> & new_added_disks)
+{
+    if (!data_volume)
+        return true;
+
+    for (auto & disk : data_volume->getDisks())
```

Comment:
But does this make a lot of sense? I mean it only works if disk had been included into the volume initially, otherwise it is no-op. What is the use-case for this?

Reply:
> I mean it only works if disk had been included into the volume initially

You're right, we need to re-initialize `storage_policy` and `data_volume` first.